crates/rover-client/src/shared/composition_error.rs

```diff
@@ -16,3 +22,74 @@ impl Display for CompositionError {
         write!(f, "{}", &self.message)
     }
 }
+
+#[derive(Debug, Default, Serialize, Clone, PartialEq)]
+pub struct CompositionErrors {
+    composition_errors: Vec<CompositionError>,
+}
+
+impl CompositionErrors {
+    pub fn new() -> Self {
+        CompositionErrors {
+            composition_errors: Vec::new(),
+        }
+    }
+
+    pub fn len(&self) -> String {
+        let num_failures = self.composition_errors.len();
+        if num_failures == 0 {
+            unreachable!("No composition errors were encountered while composing the supergraph.");
```

Comment:
unreachable technically forwards directly to a panic, are you ok with that for this particular case?

Reply:
I think so! I want folks to report a bug if this is ever executed.